runtime-decompiler/src/main/java/org/jrd/backend/data/ArchiveManager.java

```diff
@@ -70,8 +71,17 @@ private boolean findClazz(ZipInputStream zis, String clazz) throws IOException {
         return false;
     }
 
-    public static boolean shouldOpen(ZipInputStream zis) throws IOException {
-        return zis.getNextEntry() != null;
+    /**
+     * Determines whether this file can be opened with ZipInputStream
+     * @param n Name of the file
+     * @return Whether file can opened with ZipInputStream
+     */
+    public static boolean shouldOpen(String n) throws IOException {
+        /* This way has been selected as there's no other "easier" way of determining if it is an archive.
+         * If you wish to add a format, feel free to PR */
+        // return name.endsWith(".zip") || name.endsWith(".zipx") || name.endsWith(".zz") || name.endsWith(".jar") || name.endsWith(".a") || name.endsWith(".ar") || name.endsWith(".cpio") || name.endsWith(".shar") || name.endsWith(".lbr") || name.endsWith(".iso") || name.endsWith(".mar") || name.endsWith(".sbx") || name.endsWith(".tar");
```

Comment:
pls include the ```return zis.getNextEntry() != null;``` nad describne the ssue we had with it and what we debugged

Reply:
Should the previous check be added as a commentary with a description of what was not working and probable cause?